### PR TITLE
REST & API: uniform case for urls in flask subscription endpoint Fix: #6316

### PR DESCRIPTION
**Pull request by voetberg**
Addressing #6316 

As stated in the comment there - this is almost certainly a minor mistake. Alt solution would be setting up a automatic redirect for variants of urls with mixed case - which would be a larger undertaking and a separate issue.

### CODE REVIEW

**Review by bari12 — CHANGES_REQUESTED** [review]
I think this will break backwards compatibility. Because you might still have old clients (pre your change) going against the upper-case endpoints. And in this case, the server will reply a 404. 

We probably have to add rules for both, the lower-case endpoint and the previous uppercase endpoint and take it out once the `33` clients fall out of the legacy window. (Which will be the Rucio `38` release)

File: lib/rucio/web/rest/flaskapi/v1/subscriptions.py [hunk 1]
```diff
@@ -622,17 +622,23 @@ def blueprint():
...
-    bp.add_url_rule('/<account>/<name>/Rules', view_func=rules_view, methods=['get', ])
+    bp.add_url_rule('/<account>/<name>/rules', view_func=rules_view, methods=['get', ])
     subscription_view = Subscription.as_view('subscription')
     bp.add_url_rule('/<account>/<name>', view_func=subscription_view, methods=['get', 'post', 'put'])
     bp.add_url_rule('/<account>', view_func=subscription_view, methods=['get', ])
     bp.add_url_rule('/', view_func=subscription_view, methods=['get', ])
     subscription_name_view = SubscriptionName.as_view('subscription_name')
+    bp.add_url_rule('/name/<name>', view_func=subscription_name_view, methods=['get', ])
+
+    # Legacy url support
```
Comment (bari12) [inline lib/rucio/web/rest/flaskapi/v1/subscriptions.py:634]: Please add a comment:
`TODO: Take out with Rucio 38` :-)

File: lib/rucio/web/rest/flaskapi/v1/subscriptions.py [hunk 2]
```diff
@@ -622,17 +622,24 @@ def blueprint():
...
+    # TODO: Take out with Rucio 38
+    bp.add_url_rule('/Id/<subscription_id>', view_func=subscription_id_view, methods=['get', ])
+    bp.add_url_rule('/<account>/<name>/Rules/States', view_func=states_view, methods=['get', ])
+    bp.add_url_rule('/<account>/<name>/Rules', view_func=rules_view, methods=['get', ])
```
Comment (bari12) [inline lib/rucio/web/rest/flaskapi/v1/subscriptions.py:639]: I think one legacy rule is missing
`/<account>/Rules/States`

**Review by bari12 — CHANGES_REQUESTED** [review]
One last change 😄

**Comment by voetberg** [comment]
Test failure due to infamous download_pfns - otherwise everything in the clear :o

**Comment by bari12** [comment]
Please rebase and let's run the tests again.

**Review by bari12 — APPROVED** [review]

**Pull request merged**